libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -0,0 +1,337 @@
+   curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+   curl_easy_setopt(curl, CURLOPT_POSTFIELDS, bulking.c_str());
+   curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+   curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *)&readBuffer);
+   curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcrp/0.1");
+   //curl_easy_setopt(curl, CURLOPT_VERBOSE, true);
+   curl_easy_perform(curl);
+
+   long http_code = 0;
+   curl_easy_getinfo (curl, CURLINFO_RESPONSE_CODE, &http_code);
+   if(http_code == 200) {
+      // all good, do nothing
+   }
+   else if(http_code == 429) {
+      // repeat request?
+   }
+   else {
+      // exit everything ?
+   }
+   if(_es_objects_logs) {
+      auto logs = readBuffer;
+      // do logs
+      std::string url_logs = _es_objects_elasticsearch_url + "objects_logs/data/";
+      curl_easy_setopt(curl, CURLOPT_URL, url_logs.c_str());
+      curl_easy_setopt(curl, CURLOPT_POST, true);
+      curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+      curl_easy_setopt(curl, CURLOPT_POSTFIELDS, logs.c_str());
+      curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+      curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *) &readBuffer_logs);
+      curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcrp/0.1");
+      //curl_easy_setopt(curl, CURLOPT_VERBOSE, true);
+      //ilog("log here curl: ${output}", ("output", readBuffer_logs));
+      curl_easy_perform(curl);
+
+      http_code = 0;
+      curl_easy_getinfo (curl, CURLINFO_RESPONSE_CODE, &http_code);
+      if(http_code == 200) {
+         // all good, do nothing
+      }
+      else if(http_code == 429) {
+         // repeat request?
+      }
+      else {
+         // exit everything ?
+      }
+   }
+}
+
+void es_objects_plugin_impl::PrepareProposal(const proposal_object* proposal_object)
+{
+   proposal_struct prop;
+   prop.id = proposal_object->id;
+   prop.expiration_time = proposal_object->expiration_time;
+   prop.review_period_time = proposal_object->review_period_time;
+   prop.proposed_transaction = fc::json::to_string(proposal_object->proposed_transaction);
+
+   std::string data = fc::json::to_string(prop);
+
+   createBulk("proposal", data, fc::json::to_string(proposal_object->id));
+}
+
+void es_objects_plugin_impl::PrepareAccount(const account_object* account_object)
+{
+   account_struct acct;
+   acct.id = account_object->id;
+   acct.membership_expiration_date = account_object->membership_expiration_date;
+   acct.registrar = account_object->registrar;
+   acct.referrer = account_object->referrer;
+   acct.lifetime_referrer = account_object->lifetime_referrer;
+   acct.network_fee_percentage = account_object->network_fee_percentage;
+   acct.lifetime_referrer_fee_percentage = account_object->lifetime_referrer_fee_percentage;
+   acct.referrer_rewards_percentage = account_object->referrer_rewards_percentage;
+   acct.name = account_object->name;
+   acct.owner = fc::json::to_string(account_object->owner.get_addresses());
```

Comment:
added in this commit https://github.com/bitshares/bitshares-core/pull/500/commits/3545c22ab806bcb7b2d060da6b6260f4a9eda7c3
i had to add the whole `flat_map` in text form to make it work. this will make the indexing by this fields hard but i think its ok, the real intention is to have something available to relate the `accounts` index with the `balances` that uses an `owner` key.